internal/handler/mqtthandler/mqtt_handler.go

```diff
@@ -48,13 +48,12 @@ func NewHandler(server, username, password, cafile string) (handler.Handler, err
 	opts.SetOnConnectHandler(h.onConnected)
 	opts.SetConnectionLostHandler(h.onConnectionLost)
 
-	if cafile != "" {
-		tlsconfig, err := newTLSConfig(cafile)
-		if err != nil {
-			log.Fatalf("Error with the mqtt CA certificate: %s", err)
-		} else {
-			opts.SetTLSConfig(tlsconfig)
-		}
+	tlsconfig, err := newTLSConfig(cafile, certFile, certKeyFile)
+	if err != nil {
+		log.Fatalf("Error with the mqtt CA certificate: %s", err)
```

Comment:
Maybe this would print more debugging information:

```go
log.WithError(err).WithFields(log.Fields{
    "ca_cert": cafile,
    "tls_cert": certFile,
    "tls_key": certKeyFile,
}).Fatal("error loading mqtt certificate files")
```